nemo/collections/nlp/models/language_modeling/megatron/gpt_model.py

```diff
@@ -308,6 +315,12 @@ def forward(
                 sequence_parallel=self.sequence_parallel,
                 gradient_accumulation_fusion=self.config.gradient_accumulation_fusion,
             )
+            if loss_mask is not None:
```

Comment:
`post_process_result` is a tuple of `(loss, logits)` when `encoder_input is not None`, which will break the line `res = torch.zeros_like(labels).type_as(post_process_result)`. Maybe consider this:

```python
if isinstance(post_process_result, tuple):
    loss, logits = post_process_result
 else:
    loss, logits = post_process_result, None
 
if loss_mask is not None:
    res = torch.zeros_like(labels).type_as(loss)
    res[loss_mask == 1] = loss
    if logits is not None:
        return res, logits
     else:
        return res
else:
    return post_process_result

Reply:
good catch!

Reply:
we should consider a CI test when encoder_input is not None.